projects/igniteui-angular/src/lib/directives/text-selection/text-selection.directive.spec.ts

```diff
@@ -72,39 +75,41 @@ describe('IgxSelection', () => {
         const inputNativeElem = input.nativeElement;
         const inputElem: HTMLElement = input.nativeElement;
 
-        selectableTypes.forEach( el => {
-            let type = Object.keys(el)[0];
-            let val = el[type];
-            fix.componentInstance.inputType = type;
-            fix.componentInstance.inputValue = val;
-            fix.detectChanges();
-
-            inputElem.click();
-            fix.detectChanges();
-
-            if(type !== 'number'){
-                expect(inputNativeElem.selectionEnd).toEqual(inputNativeElem.value.length);
-                expect(inputNativeElem.value.substring(inputNativeElem.selectionStart, inputNativeElem.selectionEnd))
-                    .toEqual(val);
-            }
-
-            if(type === 'number'){
-                let selection = document.getSelection().toString();
-                tick(1000);
-                expect((String(val)).length).toBe(selection.length);
-            }
-        });
-
-        nonSelectableTypes.forEach( el => {
-            let type = Object.keys(el)[0];
-            let val = el[type];
-            fix.componentInstance.inputType = type;
-            fix.componentInstance.inputValue = val;
-            fix.detectChanges();
-
-            inputElem.focus();
-            fix.detectChanges();
-            expect(inputNativeElem.selectionStart).toEqual(inputNativeElem.selectionEnd);
+        requestAnimationFrame(() => {
```

Comment:
And this one is already `fakeAsync` so no need to wrap w/ rAF when you can tick already